source/application/SXRMB/SXRMB.h

```diff
@@ -43,7 +43,8 @@ namespace SXRMB {
 	{
 		NoXRFDetector = 0,
 		BrukerDetector = 1,
-		FourElementDetector = 2
+		FourElementDetector = 2,
+		BrukerAndFourElementDetector = 3
```

Comment:
By having these as QFlags, you don't need to add this enum like this.  Instead you can do:

```c++
QFlag flag = detectorFlag();

if (flag.testFlag(SXRMB::FourElement) && flag.testFlag(SXRMB::Bruker))
   // Do stuff
```

Then, all you need to do is make sure your enum is always a binary sequence then you can use logic + testFlag() for everything.